packages/core/src/app/utils/range.ts

```diff
@@ -65,6 +65,39 @@ export const isRangeSubset = (subset: Range, superset: Range): boolean => {
   )
 }
 
+/**
+ * Determines whether the provided Ranges collectively span the Range in question
+ *
+ * @param ranges the Ranges that will/won't span the rangeToSpan
+ * @param rangeToSpan the Range being spanned
+ * @returns true if ranges span rangeToSpan, false otherwise
+ */
+export const rangesSpanRange = (
+  ranges: Range[],
+  rangeToSpan: Range
+): boolean => {
+  const sortedRanges: Range[] = ranges.sort((a: Range, b: Range) => {
+    return a.start.lt(b.start) ? -1 : a.start.eq(b.start) ? 0 : 1
+  })
+
+  let spannedEnd: BigNum = rangeToSpan.start
```

Comment:
'Spannedend' as well I'm struggling to understand the meaning here

Reply:
Good call -- I updated the name of this var and added some comments and examples to the function to make it a bit clearer what it's trying to do.